tensorboard/webapp/app_routing/route_config_test.ts

```diff
@@ -503,4 +502,103 @@ describe('route config', () => {
       );
     });
   });
+
+  describe('programmatical redirection', () => {
+    it('redirects based on returned value on redirector', () => {
+      const config = new RouteConfigs([
+        {
+          path: '/tensor/:eid',
+          redirector: (paths) => {
+            const eidPart = paths[1];
+            return eidPart === '123' ? ['board'] : ['rocks'];
+          },
+        },
+        buildConcreteRouteDef({
+          routeKind: RouteKind.EXPERIMENT,
+          path: '/rocks',
+        }),
+        buildConcreteRouteDef({
+          routeKind: RouteKind.COMPARE_EXPERIMENT,
+          path: '/board',
+        }),
+      ]);
+
+      expect(config.match(buildNavigation({pathname: '/tensor/123'}))).toEqual(
+        buildRouteMatch({
+          routeKind: RouteKind.COMPARE_EXPERIMENT,
+          pathname: '/board',
+          params: {},
+        })
+      );
+      expect(config.match(buildNavigation({pathname: '/tensor/6006'}))).toEqual(
+        buildRouteMatch({
+          routeKind: RouteKind.EXPERIMENT,
+          pathname: '/rocks',
+          params: {},
+        })
+      );
+    });
+
+    it('redirects multiple times', () => {
+      const config = new RouteConfigs([
+        {
+          path: '/tb/:eid',
+          redirector: (paths) => {
+            return [paths[1]];
+          },
+        },
+        buildRedirectionRouteDef({
+          path: '/redirect',
+          redirectionPath: '/tensorboard',
+        }),
+        buildConcreteRouteDef({path: '/tensorboard'}),
+      ]);
+
+      expect(config.match(buildNavigation({pathname: '/tb/redirect'}))).toEqual(
+        buildRouteMatch({
+          pathname: '/tensorboard',
+          params: {},
+        })
+      );
+    });
+
+    it('handles parameter is opaque string in path parts', () => {
```

Comment:
nit: 'handles an opaque parameter in path parts'

Reply:
Wow, that was a crappy wording. Thanks for catching that and suggesting a fix.